### PR TITLE
etcd: Move etcd status check into the background

### PR DESCRIPTION
**Pull request by tgraf**
The current etcd Status() check was performed in a blocking manner every time
`cilium status` was invoked. If any etcd endpoint was unreachable, the status
call would block until the timeout was reached.

The timeout on the etcd status check exceeded the timeout of the k8s liveness
probe which caused Cilium to be declared unheahlthy if any etcd endpoint was
unreachable.

Move the etcd status check into the background and perform it in intervals (5
seconds). The cilium status call can be made completely non-blocking with this.

Also only consider the etcd component in error state if none of the etcd
endpoints can be reached.

Signed-off-by: Thomas Graf <thomas@cilium.io>

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/kvstore/etcd.go [hunk 1]
```diff
@@ -606,19 +620,72 @@ reList:
...
+
+	return str, nil
+}
+
+func (e *etcdClient) statusChecker() error {
+	for {
+		newStatus := []string{}
+		ok := 0
+
+		log.Debugf("Performing status check to etcd")
```
Comment (aanm) [inline pkg/kvstore/etcd.go:658]: I think you can remove this debug message

**Review by aanm — COMMENTED** [review]
Wouldn't this be better of with a controller?

**Review by aanm — APPROVED** [review]

File: pkg/kvstore/etcd.go (hunk 1 above)
Comment (tgraf) [reply pkg/kvstore/etcd.go:658]: I went back and forth. I kept it so we can ensure when we last checked health of a particular etcd connection.

**Review by joestringer — APPROVED** [review]
LGTM. Seems like worst case the status might not be updated for `5s + N*5s` where N is the number of etcd nodes. This seems fine though.

**Pull request merged**